server/middleware/authentication.js

```diff
@@ -371,3 +372,32 @@ export function mustBeLoggedIn(req, res, next) {
     }
   });
 }
+
+/**
+ * Checks if the user has two-factor authentication enabled on their account,
+ * and if they do, we challenge them to auth with it during the login flow
+ */
+export const twoFactorAuthCheck = (req, res, next) => {
+  const { TOTPCode } = req.body;
+  const { updateToken } = res.locals;
+  const email = req.remoteUser.email;
+  let twoFactorAuthToken, verified;
+  return User.findOne({ where: { email: email.toLowerCase() } })
+    .then(u => {
```

Comment:
`async`/`await` could be cleaner than promises here. In any cases it seems like these 3 `.then` could fit in just one.